test/interceptors/cache.js

```diff
@@ -1513,23 +1570,87 @@ describe('Cache Interceptor', () => {
     const request = {
       origin: 'localhost',
       method: 'GET',
-      path: '/'
+      path: '/',
+      headers: {
+        range: 'bytes=10-'
+      }
     }
 
     // First request should hit the origin
     {
       const res = await client.request(request)
       equal(requestsToOrigin, 1)
-      equal(res.statusCode, 418)
-      strictEqual(await res.body.text(), 'I am a teapot')
+      equal(res.statusCode, code)
+      strictEqual(await res.body.text(), body)
     }
 
-    // Second request should also hit the origin (not cached)
+    // Second request with different range should hit the origin too
+    request.headers.range = 'bytes=5-'
     {
       const res = await client.request(request)
-      equal(requestsToOrigin, 2) // Should be 2 (not cached)
-      equal(res.statusCode, 418)
-      strictEqual(await res.body.text(), 'I am a teapot')
+      equal(requestsToOrigin, 2)
+      equal(res.statusCode, code)
+      strictEqual(await res.body.text(), body)
+    }
+  })
+
+  test('discriminates caching of conditionnal requests, or does not cache them', async () => {
```

Comment:
This test needs to be split in two, because `if-none-match` causes `if-modified-since` to be ignored. So, its third test could wrongly fail.